packages/builders/src/components/v2/Section.ts

```diff
@@ -0,0 +1,257 @@
+
+		this.data.accessory = builder;
+		return this;
+	}
+
+	/**
+	 * Sets a danger button component to be the accessory of this section.
+	 *
+	 * @param input - The button to set as the accessory
+	 */
+	public setDangerButtonAccessory(
+		input:
+			| DangerButtonBuilder
+			| ((builder: DangerButtonBuilder) => DangerButtonBuilder)
+			| (APIButtonComponentWithCustomId & { style: ButtonStyle.Danger }),
+	): this {
+		const builder = resolveBuilder(input, DangerButtonBuilder);
+
+		this.data.accessory = builder;
+		return this;
+	}
+
+	/**
+	 * Sets a SKU id button component to be the accessory of this section.
+	 *
+	 * @param input - The button to set as the accessory
+	 */
+	public setPremiumButtonAccessory(
+		input:
+			| APIButtonComponentWithSKUId
+			| PremiumButtonBuilder
+			| ((builder: PremiumButtonBuilder) => PremiumButtonBuilder),
+	): this {
+		const builder = resolveBuilder(input, PremiumButtonBuilder);
+
+		this.data.accessory = builder;
+		return this;
+	}
+
+	/**
+	 * Sets a URL button component to be the accessory of this section.
+	 *
+	 * @param input - The button to set as the accessory
+	 */
+	public setLinkButtonAccessory(
+		input: APIButtonComponentWithURL | LinkButtonBuilder | ((builder: LinkButtonBuilder) => LinkButtonBuilder),
+	): this {
+		const builder = resolveBuilder(input, LinkButtonBuilder);
+
+		this.data.accessory = builder;
+		return this;
+	}
+
+	/**
+	 * Sets a thumbnail component to be the accessory of this section.
+	 *
+	 * @param input - The thumbnail to set as the accessory
+	 */
+	public setThumbnailAccessory(
+		input: APIThumbnailComponent | ThumbnailBuilder | ((builder: ThumbnailBuilder) => ThumbnailBuilder),
+	): this {
+		const builder = resolveBuilder(input, ThumbnailBuilder);
+
+		this.data.accessory = builder;
+		return this;
+	}
+
+	/**
+	 * Removes, replaces, or inserts text display components for this section.
+	 *
+	 * @param index - The index to start removing, replacing or inserting text display components
+	 * @param deleteCount - The amount of text display components to remove
+	 * @param components - The text display components to insert
+	 */
+	public spliceTextDisplayComponents(
+		index: number,
+		deleteCount: number,
+		...components: RestOrArray<
+			APITextDisplayComponent | TextDisplayBuilder | ((builder: TextDisplayBuilder) => TextDisplayBuilder)
+		>
+	): this {
+		const normalized = normalizeArray(components);
+		const resolved = normalized.map((component) => resolveBuilder(component, TextDisplayBuilder));
+
+		this.data.components.splice(index, deleteCount, ...resolved);
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc ComponentBuilder.toJSON}
+	 */
+	public override toJSON(validationOverride?: boolean): APISectionComponent {
+		const { components, accessory, ...rest } = this.data;
+
+		const data = {
+			...structuredClone(rest),
+			components: components.map((component) => component.toJSON(validationOverride)),
```

Comment:
```suggestion
			components: components.map((component) => component.toJSON(false)),
```

`accessory` is correct since `sectionPredicate`'s `accessory` only validates the `type`